app/src/androidTest/java/at/tugraz/ist/swe/photogallery/BasicGalleryInstrumentedTest.java

```diff
@@ -45,4 +49,40 @@ public void checkGalleryIsDisplayed() {
         onView(withId(R.id.gallery)).check(matches(isDisplayed()));
     }
 
+
+    public static void assertViewWithTextIsVisible(UiDevice device, String text) {
+        UiObject allowButton = device.findObject(new UiSelector().text(text));
+        if (!allowButton.exists()) {
+            throw new AssertionError("View with text <" + text + "> not found!");
+        }
+    }
+
+
+    public static void denyCurrentPermission(UiDevice device) throws UiObjectNotFoundException {
+        UiObject denyButton = device.findObject(new UiSelector().text("ABLEHNEN"));
+        denyButton.click();
+    }
+
+    @Test
+    public void shouldDisplayPermissionRequestDialogAtStartup() throws Exception {
+        UiDevice device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation());
+        assertViewWithTextIsVisible(device, "ZULASSEN");
```

Comment:
Please retrieve string from system using them locale/language, as this will only work with system language set to german.